File: backend/app/schemas/catalog.py
import uuid

from pydantic import BaseModel, ConfigDict


class CityOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    slug: str
    is_municipality: bool


class CategoryOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    slug: str


class CategoryGroupOut(BaseModel):
    """Trả về đúng dạng cây như category_groups trong data.json mẫu."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    group_name: str
    group_slug: str
    categories: list[CategoryOut]

File: frontend/src/types/catalog.ts
export interface City {
  id: number
  name: string
  slug: string
  is_municipality: boolean
}

export interface Category {
  id: string
  name: string
  slug: string
}

export interface CategoryGroup {
  id: string
  group_name: string
  group_slug: string
  categories: Category[]
}
